### PR TITLE
[Fix] Fix bug in visualize_smpl and smpl data

### PR DESCRIPTION
**Pull request by Yamato-01**
1. Fix the correspondence of  mask's batchsize in `smpl-data.py`.
2. Fix running long-time video-writing running in `visualize_smpl.py`. Fix the absence of `images_to_video `function

### CODE REVIEW

File: xrmocap/core/visualization/visualize_smpl.py [hunk 1]
```diff
@@ -141,41 +142,54 @@ def visualize_smpl_data(
...
-        end_idx = min(start_idx + batch_size, data_len)
-        # prepare background array for this batch
+    # check whether to write video or write images
+    write_video, write_img = (True, False) \
+        if check_path_suffix(output_path, '.mp4') else (False, True)
+    if write_video:
+        xrprimer_video_writer = VideoWriter(
+            output_path, [cam_param.height, cam_param.width])
+
+    total_iter = data_len // batch_size + 1
```
Comment (LazyBusyYang) [inline xrmocap/core/visualization/visualize_smpl.py:152]: What if data_len=10 and batch_size=5? I suggest something like `math.ceil()`.

File: xrmocap/core/visualization/visualize_smpl.py [hunk 2]
```diff
@@ -141,41 +142,54 @@ def visualize_smpl_data(
...
+
+            param_dict_input = dict()
+            for key, value in param_dict.items():
+                if key not in ['right_hand_pose', 'left_hand_pose']:
+                    param_dict_input[key] = torch.tensor(
+                        value[start_idx:end_idx, :],
+                        device=device,
+                        dtype=torch.float32)
+                else:
+                    param_dict_input[key] = torch.tensor(
```
Comment (LazyBusyYang) [inline xrmocap/core/visualization/visualize_smpl.py:172]: Wrong hands shape has been fixed in https://github.com/openxrlab/xrmocap/pull/88. Shape of `param_dict['right_hand_pose']` is `(n_frame, 45)` now.

File: xrmocap/core/visualization/visualize_smpl.py [hunk 3]
```diff
@@ -141,41 +142,54 @@ def visualize_smpl_data(
...
+        start_idx = curr_iter * batch_size
+        end_idx = min(data_len, start_idx + batch_size)
+        for person_idx in range(n_person):
+            smpl_data = smpl_data_list[person_idx]
+            param_dict = smpl_data.to_tensor_dict(device=device)
+
+            param_dict_input = dict()
+            for key, value in param_dict.items():
+                if key not in ['right_hand_pose', 'left_hand_pose']:
+                    param_dict_input[key] = torch.tensor(
```
Comment (LazyBusyYang) [inline xrmocap/core/visualization/visualize_smpl.py:167]: `dtype` and `device` have already been converted at https://github.com/openxrlab/xrmocap/blob/main/xrmocap/data_structure/body_model/smplx_data.py#L380. This loop can be removed.

File: xrmocap/core/visualization/visualize_smpl.py [hunk 4]
```diff
@@ -196,11 +210,13 @@ def visualize_smpl_data(
...
-                shape=(end_idx - start_idx, cam_param.height, cam_param.width,
-                       3),
+                (end_idx - start_idx, cam_param.height, cam_param.width, 3),
                 dtype=np.uint8)
         batch_results = []
-        for frame_idx in range(start_idx, end_idx):
+
+        start_idx = 0
+        end_idx = min(data_len - curr_iter * batch_size, batch_size)
+        for frame_idx in tqdm(range(start_idx, end_idx), disable=disable_tqdm):
```
Comment (LazyBusyYang) [inline xrmocap/core/visualization/visualize_smpl.py:219]: What about we do not re-assign values of `start_idx` and `end_idx`, but edit for loop arguments:
```suggestion
        for relative_idx in tqdm(range(0, end_idx-start_idx), disable=disable_tqdm):
```

File: xrmocap/core/visualization/visualize_smpl.py [hunk 5]
```diff
@@ -141,41 +142,54 @@ def visualize_smpl_data(
...
+                body_model_output = model(**param_dict_input)
+            verts = body_model_output['vertices']  # n_batch, n_verts, 3
+            sperson_verts = torch.unsqueeze(verts, dim=1)
+            # sperson_verts.shape: n_batch, n_person, n_verts, 3
+            if mperson_verts is None:
+                mperson_verts = sperson_verts
+            else:
+                mperson_verts = torch.cat((mperson_verts, sperson_verts),
+                                          dim=1)
+        file_names_cache = None
```
Comment (LazyBusyYang) [inline xrmocap/core/visualization/visualize_smpl.py:190]: Is it in use? It makes sense only when `write_img is True`.

File: xrmocap/data_structure/body_model/smpl_data.py [hunk 6]
```diff
@@ -486,6 +489,7 @@ def load(self, npz_path: str):
         """
         with np.load(npz_path, allow_pickle=True) as npz_file:
             tmp_data_dict = dict(npz_file)
+
```
Comment (LazyBusyYang) [inline xrmocap/data_structure/body_model/smpl_data.py:492]: We do not need blank line in such a short function. Add new line only when it's too long to read.

**Review by LazyBusyYang — CHANGES_REQUESTED** [review]
Something needs to be modified as the main branch was updated yesterday.

**Comment by LazyBusyYang** [comment]
This PR is replaced by https://github.com/openxrlab/xrmocap/pull/89.
